### PR TITLE
ARINA read

### PR DESCRIPTION
**Pull request by smribet**

### CODE REVIEW

**Comment by smribet** [comment]
This PR includes 
- two new file readers: one for abTEM data and one for arina data 
- a h5 file parser 
- reformatting of some existing files

Reading 2D calibrations requires the patches in PR 462

File: py4DSTEM/process/virtualimage.py
Comment (bsavitzky) [inline py4DSTEM/process/virtualimage.py:1]: same comment as for virtualdiffraction.py

**Review by bsavitzky — CHANGES_REQUESTED** [review]
Looks good overall!  Just a few things that could use editing.  There are going to be merge conflicts, I can take a stab at handling those and building tests for the new readers

File: py4DSTEM/process/virtualdiffraction.py
[the remark this answers is not in the document]
Comment (smribet) [reply py4DSTEM/process/virtualdiffraction.py:1]: just autoformatter

File: py4DSTEM/io/filereaders/read_abTEM.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            return diffraction
+        elif units[0] == b"\xc3\x85":
+            image = VirtualImage(data=data)
+            if sampling[0] != sampling[1]:
+                print(
+                    "Warning: py4DSTEM currently only handles uniform x,y sampling. Setting sampling with x calibration"
+                )
+            image.calibration.set_Q_pixel_units("A")
+            image.calibration.set_Q_pixel_size(sampling[0])
+            return image
```
[the remark this answers is not in the document]
Comment (bsavitzky) [reply py4DSTEM/io/filereaders/read_abTEM.py:80]: Cool thanks - noticing now that an `else` statement is also needed in the nested if/else clauses one level in - perhaps line 72 can just be changed from `elif` to `else`?  I'm going to be doing merge conflicts now, I can just roll that in there

[the remark this answers is not in the document]
Comment (smribet) [reply py4DSTEM/io/filereaders/read_abTEM.py:80]: yes that should work! Thanks Ben!

**Comment by bsavitzky** [comment]
Ok, I've merged the dev head into this branch, fixed consequent merge conflicts and import errors, added the arina test file to the test suite test file downloader with the rest of the io test bundle, and added a unit test file for the arina reader.  So the infrastructure for the arina file reader should be in place.  Do have have one or two test files you'd like to use for the abTEM reader?  (two if you'd like to test both the 4D and 2D use cases).  If so we can add those to the test suite as well.

When I ran the arina reader I hit an error - here's the traceback:

```
    def test_read_arina():

        # read
>       data = py4DSTEM.import_file( filepath )

test_arina.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../py4DSTEM/io/importfile.py:93: in import_file
    data = read_arina(filepath, mem=mem, binfactor=binfactor, **kwargs)
../../py4DSTEM/io/filereaders/read_arina.py:40: in read_arina
    nimages = nimages + f["entry"]["data"][dset].shape[0]
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
/Users/Ben/opt/anaconda3/envs/py4dstem_dev/lib/python3.10/site-packages/h5py/_hl/group.py:357: in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   KeyError: "Unable to open object (unable to open external file, external link file name = 'STO_STEM_bench_20us_data_000001.h5')"

h5py/h5o.pyx:190: KeyError
```

Seems like it found and read the master file but then had trouble reading the first data file!

**Comment by smribet** [comment]
Thanks Ben! I'm having trouble recreating that error you posted. This just worked for me: 
```
test = py4DSTEM.import_file('STO_STEM_bench_20us_master.h5', scan_width = 1000)
test

DataCube( A 4-dimensional array of shape (1000, 1000, 96, 96) called 'datacube',
          with dimensions:

          Rx = [0,1,...] pixels
          Ry = [0,1,...] pixels
          Qx = [0,1,...] pixels
          Qy = [0,1,...] pixels

```

**Comment by smribet** [comment]
Added the correction option, so I think this should be ready now

**Review by bsavitzky — APPROVED** [review]

**Pull request merged**